src/puppetlabs/puppetdb/http/handlers.clj

```diff
@@ -53,17 +55,44 @@
    route-param-key :- s/Keyword]
   (cmdi/wrap-routes route #(parent-check % version entity route-param-key)))
 
+(defn wrap-apply-active-node-restriction [handler]
+  (fn [req]
+    (cond
+      (some-> req :puppetdb-query :ast_only http-q/coerce-to-boolean) (handler req)
+      (= ["from" "packages"] (some->> req :puppetdb-query :query (take 2))) (handler req)
+      :else (handler (http-q/restrict-query-to-active-nodes req)))))
+
+(defn req-body->stream [req req-body-map]
```

Comment:
...also, minor, but these guards are complex enough that I might be inclined to pull the bodies onto their own lines:
```clojure
(cond
  (some-> req :puppetdb-query :ast_only http-q/coerce-to-boolean)
  (handler req)
  (= ["from" "packages"] (some->> req :puppetdb-query :query (take 2)))
  (handler req)
  :else (handler (http-q/restrict-query-to-active-nodes req)))
```